packages/cli/src/runAndroid/runAndroid.js

```diff
@@ -191,6 +200,32 @@ function tryInstallAppOnDevice(args, device) {
   }
 }
 
+function getInstallApkName(
+  appFolder,
+  adbPath,
+  variant,
+  device,
+  buildDirectory
+) {
+  const availableCPUs = adb.getAvailableCPUs(adbPath, device);
+
+  // check if there is an apk file like app-armeabi-v7a-debug.apk
+  for (const availableCPU of availableCPUs.concat('universal')) {
+    const apkName = `${appFolder}-${availableCPU}-${variant}.apk`;
+    if (fs.existsSync(`${buildDirectory}/${apkName}`)) {
+      return apkName;
+    }
+  }
+
+  // check if there is a default file like app-debug.apk
+  const apkName = `${appFolder}-${variant}.apk`;
+  if (fs.existsSync(`${buildDirectory}/${apkName}`)) {
+    return apkName;
+  }
+
+  throw new Error('Not found the correct install APK file!');
```

Comment:
What about adding a "last resort" function to find apk by using unix `find` (or windows - [dir](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/dir)) to search `appFolder` for all `*.apk` existance?

Reply:
@Krizzu I don't think I need the logic you said. I'm sure I checked all available files in my previous logic. Please let me know if there is an exception case.

Reply:
Just thought about product flavors, because they change output directories when used

Reply:
@Krizzu `flavor` has been deprecated. Use `variant` instead. This option works properly with the source I worked on.

Reply:
@ifsnow I'm talking about `product flavor` of Android plugin - if you have those in your build, you'll get different output dirs for your APKs. It's not the case in default RN project tho, so I think we can pass for now.